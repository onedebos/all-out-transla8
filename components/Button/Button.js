import styles from './Button.module.css'

export default function Button({loading}){

    return (
        <button class={styles.button} data-testid="submit">{loading ? "Translating..." : "Translate"}</button>
    )
}