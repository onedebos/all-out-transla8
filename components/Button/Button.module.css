.button{
    width: 300px;
    background-color:#ddd;
    color: #000;
    padding: 10px 0;
    border-radius: 10px;
    font-weight: 600;
    text-align: center;
    transition: 0.3s;
}

.button:hover{
    background-color: rgb(66, 133, 244);
    color: #fff;
    transition: 0.3s;
}