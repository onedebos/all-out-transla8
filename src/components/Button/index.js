import styled from 'styled-components';

export default function Button({ loading = false, onClick }) {
	return (
		<ButtonStyles onClick={onClick} loading={loading ? 1 : 0} data-testid="translate">
			{loading ? 'Translating....' : 'Translate'}
		</ButtonStyles>
	);
}

const ButtonStyles = styled.button`
	width: 150px;
	background-color: ${(props) => (props.loading ? 'rgb(66, 133, 244)' : '#ddd')};
	color:${(props) => (props.loading ? '#fff' : '#000')};
	padding: 10px 0;
	border-radius: 10px;
	font-weight: 600;
	text-align: center;
	transition: 0.3s;

	&:hover {
		background-color: rgb(66, 133, 244);
		color: #fff;
		transition: 0.3s;
	}
`;
