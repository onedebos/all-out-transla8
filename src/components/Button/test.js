/**
 * @jest-environment jsdom
 */
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Button from './';

afterEach(() => {
	cleanup();
});

describe('<Button />', () => {
	it('renders the Button component successfully without crashing', () => {
		render(<Button />);
	});

	it('renders the Button element', () => {
		render(<Button />);
		const btnElement = screen.getByTestId('translate');
		expect(btnElement).toBeInTheDocument();
	});

	it('renders the correct text when loading is false', () => {
		render(<Button loading={false} />);
		const btnElement = screen.getByTestId('translate');
		expect(btnElement).toHaveTextContent('Translate');
	});


	it('has a background color of #ddd by default', () => {
		render(<Button loading={false} />);
		const btnElement = screen.getByTestId('translate');
		expect(btnElement).toHaveStyle({
			'background-color': '#ddd',
		});
	});

	it('has a background color of rgb(66, 133, 244) when loading is true', () => {
		render(<Button loading={true} />);
		const btnElement = screen.getByTestId('translate');
		expect(btnElement).toHaveStyle({
			'background-color': 'rgb(66, 133, 244)',
			color: '#fff',
		});
	});

	it('matches snapshot', () => {
		const tree = renderer.create(<Button loading={false} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
