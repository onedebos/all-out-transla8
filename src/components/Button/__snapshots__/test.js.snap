// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<button
  className="Button__ButtonStyles-sc-17n8azk-0 kbnOiR"
  data-testid="translate"
  loading={0}
>
  Translate
</button>
`;
